bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -432,24 +431,24 @@ private void createSearchTools() {
 		createAreaSearchButton();
 
 		@SuppressWarnings("unused")
-		ToolItem separator = new ToolItem(searchTools, SWT.SEPARATOR);
+		ToolItem separator = searchTools.createToolItem(SWT.SEPARATOR);
 
```

Comment:
It looks to me that you could extract the repetitive logic of creating buttons into a factory, I see the pattern:

(1. new ToolItem)
2. searchTools
3. setImage()
4. setToolTipText()
5. addSelectionListener...

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/292d692d7164d2767167f1ac58161f25dd30825e